### PR TITLE
fix #1118

### LINKED ISSUE #1118: Inconsistent search and body rendering
#### Steps to reproduce

```
## fatal error: 'openssl/ssl.h' file not found
```

#### What is current behaviour

image

The link when searching is `/question/install?id=fatal-error-opensslsslh-file-not-found`

image

The link when body is `/question/install?id=fatal-error-39opensslsslh39-file-not-found`

image

#### What is the expected behaviour

`/question/install?id=fatal-error-39opensslsslh39-file-not-found`

Or just keep it consistent

#### Other relevant information

- [x] Bug does still occur when all/other plugins are disabled?

- Your OS: 
- Node.js version: 
- npm/yarn version: 
- Browser version: 
- Docsify version: 
- Docsify plugins: 

#### Please create a reproducible sandbox 

[Edit 307qqv236](https://codesandbox.io/s/307qqv236)

### PR DESCRIPTION
**Pull request by sy-records**
**Summary**

fix #1118 

This modification will ensure that the search content and body rendering are consistent.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

---

* [ ] DO NOT include files inside `lib` directory.

### CODE REVIEW

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -25,10 +25,9 @@ function escapeHtml(string) {
     '>': '&gt;',
     '"': '&quot;',
     "'": '&#39;',
-    '/': '&#x2F;',
   };
 
-  return String(string).replace(/[&<>"'/]/g, s => entityMap[s]);
+  return String(string).replace(/[&<>"']/g, s => entityMap[s]);
```
[the remark this answers is not in the document]
Comment (sy-records) [reply src/plugins/search/search.js:31]: I modified it, is this correct?

[the remark this answers is not in the document]
Comment (sy-records) [reply src/plugins/search/search.js:31]: Do you mean you don't need to modify `escapeHtml(string) `？If yes, it is still different from the markdown in the text.

[the remark this answers is not in the document]
Comment (Koooooo-7) [reply src/plugins/search/search.js:31]: I mean if another title is `example a&b`, we can't do what u did again that removing the `&`  to make them same. 
it seems not not a good way.

[the remark this answers is not in the document]
Comment (sy-records) [reply src/plugins/search/search.js:31]: Ok, I try.

**Comment by sy-records** [comment]
@Koooooo-7 Sorry, I still keep my opinion.
Can you let other team members also come to review?
The markdown rendering of the text uses `marked`, which also replaces `& <>" '`
image
After this pr is modified, `token.text`, `markdown`, and `sidebar` will be consistent.

File: src/plugins/search/search.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sy-records) [reply src/plugins/search/search.js:30]: `&` Can be processed, it will be displayed as `amp` in the url.

**Comment by Koooooo-7** [comment]
@sy-records yep, I will look on it also. 

@anikethsaha could u review this PR about those changes.

**Comment by sy-records** [comment]
> ```js
> const matchContent =
>             '...' +
>             escapeHtml(postContent)
>               .substring(start, end)
>               .replace(regEx, `<em class="search-keyword">${keyword}</em>`) +
>             '...';
> ```
> 
> and I think u need consider on this place together also.

Is there any problem here? `/` not searchable?

**Comment by Koooooo-7** [comment]
> > ```js
> > const matchContent =
> >             '...' +
> >             escapeHtml(postContent)
> >               .substring(start, end)
> >               .replace(regEx, `<em class="search-keyword">${keyword}</em>`) +
> >             '...';
> > ```
> > 
> > 
> > and I think u need consider on this place together also.
> 
> Is there any problem here? `/` not searchable?

nah,  cuz we don't have good test now, so I hope u could make sure the changes about all the `escapeHtml` place. 
Right now I thing your changes worksworks after tested again. just make sure it.

**Comment by sy-records** [comment]
Well, I tried it without problems.
If it existed before, it will be escaped：`[test](/test)` `[test](&#x2F;test)`

**Comment by Koooooo-7** [comment]
> Well, I tried it without problems.
> If it existed before, it will be escaped：`[test](/test)` `[test](&#x2F;test)`

okay, thx again. 
LGTM. :100:

**Comment by Koooooo-7** [comment]
please , squash the commits (with clearly commit message) then we are good to merge. :+1:

**Comment by anikethsaha** [comment]
@Koooooo-7 we can do squash and merge while merging, we will add an appropriate message for it according to our commit message convention. 
if you want, you can squash and merge it now as well 👍 


I am waiting for @trusktr 's review, we can merge this after 2-3 days if he is in-active .
👍

**Pull request merged**